code/datums/components/storage/storage.dm

```diff
@@ -0,0 +1,717 @@
+	if(!force && !can_be_inserted(I, TRUE, M))
+		return FALSE
+	return handle_item_insertion(I, silent, M)
+
+/datum/component/storage/proc/show_to_ghost(mob/dead/observer/M)
+	return user_show_to_mob(M, TRUE)
+
+/datum/component/storage/proc/signal_show_attempt(mob/showto, force = FALSE)
+	return user_show_to_mob(showto, force)
+
+/datum/component/storage/proc/on_check()
+	return TRUE
+
+/datum/component/storage/proc/check_locked()
+	return locked
+
+/datum/component/storage/proc/signal_take_type(type, atom/destination, amount = INFINITY, check_adjacent = FALSE, force = FALSE, mob/user)
+	if(!force)
+		if(check_adjacent)
+			if(user)
+				if(!user.CanReach(destination) || !user.CanReach(parent))
+					return FALSE
+			else if(!destination.CanReachStorage(parent))
+				return FALSE
+	var/list/taking = typecache_filter_list(contents(), typecacheof(type))
+	if(length(taking) > amount)
+		taking.Cut(amount)
+	for(var/i in taking)
+		var/atom/movable/AM = i
+		remove_from_storage(AM, destination)
+	return TRUE
+
+/datum/component/storage/proc/remaining_space_items()
+	STORAGE_COMPONENT_GET_REAL_LOCATION
+	return max(0, max_items - real_location.contents.len)
+
+/datum/component/storage/proc/signal_fill_type(type, amount = 20, force = FALSE)
+	STORAGE_COMPONENT_GET_REAL_LOCATION
+	if(!force)
+		amount = min(remaining_space_items(), amount)
+	for(var/i in 1 to amount)
+		handle_item_insertion(new type(real_location), TRUE)
+		CHECK_TICK
+	return TRUE
+
+/datum/component/storage/proc/on_attack_hand(mob/user)
+	var/atom/A = parent
+	if(!attack_hand_interact)
+		return
+	if(user.active_storage == src && A.loc == user) //if you're already looking inside the storage item
+		user.active_storage.close(user)
+		close(user)
+		. = COMPONENT_NO_ATTACK_HAND
+		return
+
+	if(rustle_sound)
+		playsound(A, "rustle", 50, 1, -5)
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.l_store == A && !H.get_active_held_item())	//Prevents opening if it's in a pocket.
+			. = COMPONENT_NO_ATTACK_HAND
+			H.put_in_hands(A)
+			H.l_store = null
+			return
+		if(H.r_store == A && !H.get_active_held_item())
+			. = COMPONENT_NO_ATTACK_HAND
+			H.put_in_hands(A)
+			H.r_store = null
+			return
+
+	if(A.loc == user)
+		. = COMPONENT_NO_ATTACK_HAND
+		show_to(user)
+
+/datum/component/storage/proc/signal_on_pickup(mob/user)
+	var/atom/A = parent
+	update_actions()
+	for(var/mob/M in range(1, A))
+		if(M.active_storage == src)
+			close(M)
+
+/datum/component/storage/proc/signal_take_obj(atom/movable/AM, new_loc, force = FALSE)
+	if(!(AM in STORAGE_COMPONENT_ACCESS_REAL_LOCATION))
+		return FALSE
+	return remove_from_storage(AM, new_loc)
+
+/datum/component/storage/proc/signal_quick_empty(atom/loctarget)
+	return do_quick_empty(loctarget)
+
+/datum/component/storage/proc/signal_hide_attempt(mob/target)
+	return hide_from(target)
+
+/datum/component/storage/proc/on_alt_click(mob/user)
+	if(user.incapacitated() || user.CanReachStorage(src))

```

Comment:
Here's the problem. This check doesn't work because a)You forgot the ! b)CanReachStorage always returns FALSE if the depth param isn't specified.
https://github.com/kevinz000/-tg-station/blob/57162b47a1e764e3e74d808197868f0bb6a9b7da/code/_onclick/click.dm#L181

Reply:
Don't let the greytiders know or they'll be stealing things out of backpacks through windows.

Reply:
whoops sorry.